core/controlplane/config/templates/cloud-config-etcd

```diff
@@ -355,6 +377,43 @@ write_files:
     content: {{$w.GzippedBase64Content}}
   {{- end }}
 {{- end }}
+{{if and (.AmazonSsmAgent.Enabled) (ne .AmazonSsmAgent.DownloadUrl "")}}
+  - path: "/opt/bin/ssm/install-ssm-agent.sh"
+    permissions: 0700
+    content: |
+      #!/bin/bash
+      set -eo pipefail
+
+      function finish {
+        if [ "$(pwd)" == "/opt/bin/ssm" ]; then
+          popd > /dev/null 2>&1
+        fi
+      }
+
+      trap finish EXIT
+
+      pushd /opt/bin/ssm > /dev/null 2>&1
+
+      TAR_FILE=ssm.linux-amd64.tar.gz
+      CHECKSUM_FILE="${TAR_FILE}.sha1"
+
+      curl --silent -L -o ${TAR_FILE} "{{ .AmazonSsmAgent.DownloadUrl }}"
+      if [[ -f "${CHECKSUM_FILE}" ]]; then
+        ( sha1sum --status -c "${CHECKSUM_FILE}" && exit 0 ) || true
```

Comment:
This is for upgrading the agent. Sometimes you might want to upgrade your agent without running `kube-aws update`. In such a case, you can overwrite the gzipped file on the download url and then reboot EC2 instances to reinstall the agent:

1. `sha1sum --status -c "${CHECKSUM_FILE}"` will fail in this case
2. then the following lines will be executed and the agent will be overwritten and upgraded